system/nxplayer/nxplayer_mp3.c

```diff
@@ -0,0 +1,321 @@
+/****************************************************************************
+ * apps/system/nxplayer/nxplayer_mp3.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <sys/types.h>
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <nuttx/audio/audio.h>
+
+#include "system/nxplayer.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define ID3V2_BIT_MASK 0x7F
+
+/****************************************************************************
+ * Private Type Declarations
+ ****************************************************************************/
+
+const static uint16_t g_mpa_freq_tab[3] =
+{
+  44100, 48000, 32000
+};
+
+const static uint16_t g_mpa_bitrate_tab[2][3][15] =
+{
+  {
+    {
+      0, 32, 64, 96, 128, 160, 192, 224, 256, 288, 320, 352, 384, 416, 448
+    },
+    {
+      0, 32, 48, 56,  64,  80,  96, 112, 128, 160, 192, 224, 256, 320, 384
+    },
+    {
+      0, 32, 40, 48,  56,  64,  80,  96, 112, 128, 160, 192, 224, 256, 320
+    }
+  },
+  {
+    {
+      0, 32, 48, 56,  64,  80,  96, 112, 128, 144, 160, 176, 192, 224, 256
+    },
+    {
+      0,  8, 16, 24,  32,  40,  48,  56,  64,  80,  96, 112, 128, 144, 160
+    },
+    {
+      0,  8, 16, 24,  32,  40,  48,  56,  64,  80,  96, 112, 128, 144, 160
+    }
+  }
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static int nxplayer_check_mpeg(uint32_t header)
+{
+  int ret = -EINVAL;
+
+  /* header */
+
+  if ((header & 0xffe00000) != 0xffe00000)
+    {
+      return ret;
```

Comment:
return -EINVAL directly and remove line 82

Reply:
done